jobs/release/release-create/Jenkinsfile

```diff
@@ -0,0 +1,199 @@
+            ]) {
+
+                for (i = 0; i < componentConfigs.size(); i++) {
+                    def index = i
+                    def componentConfig = componentConfigs[index]
+
+                    if (componentConfig['skipImageTag']) {
+                        println("[${componentConfig['name']}] - Skipped image tagging, please ensure the release tag ${releaseTagName} exists for this component!")
+                    } else {
+                        dir(componentConfig['name']) {
+                            def headHash = sh(returnStdout: true, script: "git rev-parse --short HEAD").trim()
+                            println("[${componentConfig['name']}] - Tagging docker image ${headHash} as ${releaseTagName}")
+
+                            retry(3) {
+                                sh """
+                            skopeo copy \
+                              --src-creds ${env.QUAY_CREDS} \
+                              --dest-creds ${env.QUAY_CREDS} \
+                              docker://${componentConfig['imageUrl']}:${headHash} \
+                              docker://${componentConfig['imageUrl']}:${releaseTagName}
+                            """
+                            }
+                        }
+                    }
+                }
+            }
+
+        }
+
+        stage('Update Installation Repo') {
+            println("Update Installation Repo")
+
+            dir('installation') {
+
+                def globalVariablesFile = releaseConfig?.get('ansible')?.get('variablesFile')
+
+                for (i = 0; i < componentConfigs.size(); i++) {
+                    def index = i
+                    def componentConfig = componentConfigs[index]
+
+                    def variablesFile = componentConfig?.get('ansible')?.get('variablesFile') ?: globalVariablesFile
+                    if(!variablesFile) {
+                        variablesFile = "evals/roles/${componentConfig['ansible']['role']}/defaults/main.yml"
+                    }
+
+                    if(componentConfig['ansible']['imageTagVar']) {
+                        sh "sed -i '/^${componentConfig['ansible']['imageTagVar']}/c ${componentConfig['ansible']['imageTagVar']}: ${releaseTagName}' ${variablesFile}"
+                    }
+
+                    if(componentConfig['ansible']['gitTagVar']) {
+                        sh "sed -i '/^${componentConfig['ansible']['gitTagVar']}/c ${componentConfig['ansible']['gitTagVar']}: ${releaseTagName}' ${variablesFile}"
+                    }
+
+                }
+
+                if (gitRepoIsDirty()) {
+                    sh "git commit -a -m \"[automation] Update component versions for release ${releaseTagName}\""
+                    sh "git push origin HEAD:${releaseBranchName}"
+                }
+
+                sh "git tag ${releaseTagName}"
+                sh "git push origin ${releaseTagName}"
+            }
+        }
+
+    }
+}
```

Comment:
Hope the stages here are fairly self explanatory, but this is the basic process:

1. Pull Installation repo and check that we don't already have the release tag we are about to try and create, exit if it does.
1. Fetch all component repos as described in the components.yaml file (https://github.com/integr8ly/installation/blob/a9026074c319ecae034858c0255151bb7b71602b/COMPONENTS.yaml)
1. Create a release branch of vX.X (e.g v1.2) on the installation repo **only**. Always currently created from master as per a new minor release.
1. Create a release tag on each components git repo e.g release.1.2.0 at the HEAD of the master branch at that time
1. Create a docker image tag in the components quay repo which points the current HEAD short hash tag at a new release tag e.g 1234567 -> release-1.2.0. (Note, it is expected that something will have already created a tag in the components quay repo with a tag of the current HEADs short commit hash see https://github.com/integr8ly/3scale-operator/blob/master/.travis.yml#L29)
1. Update docker image versions and git refs as appropriate for each component in the installation repos release branch, and tag with the same release e.g. release-1.2.0

Reply:
@maleck13 FYI, you happy enough with this? ^

Reply:
Since creating this pipeline we have discussed and started looking into a slightly different way of releasing where the integreatly components won;t be treated any differently from external project components. This will mean this flow and pipeline will end up changing considerably.

We can keep it this way for now though so if the need arises before that work is complete we can at least create a new release.